app/workers/work-reaper.ts

```diff
@@ -25,48 +25,99 @@ export default class WorkReaper implements Worker {
     this.logger = config.logger;
   }
 
-  async deleteTerminalWork(notUpdatedForMinutes: number, jobStatus: JobStatus[]): Promise<void> {
-    try {
-      const workItemIds = await getWorkItemIdsByJobUpdateAgeAndStatus(
-        db, notUpdatedForMinutes, jobStatus,
-      );
-      if (workItemIds.length) {
-        const chunkedWorkItemIds = _.chunk(workItemIds, batchSize);
-        for (const workItemIdsChunk of chunkedWorkItemIds) {
-          const numItemsDeleted = await deleteWorkItemsById(db, workItemIdsChunk);
-          this.logger.info(`Work reaper removed ${numItemsDeleted} work items`);
+
+  /**
+   * Find work items that are older than notUpdatedForMinutes and delete them.
+   * @param notUpdatedForMinutes - upper limit on the duration since the last update
+   * @param jobStatus - a list of terminal job statuses
+   * @returns Resolves when the request is complete
+   */
+  async deleteTerminalWorkItems(notUpdatedForMinutes: number, jobStatus: JobStatus[]): Promise<void> {
+    let done = false;
+    let startingId = 0;
+    let totalDeleted = 0;
+    const batchSize = env.workReaperBatchSize;
+    this.logger.info('Work reaper delete terminal work items started.');
+
+    while (!done) {
+      try {
+        const workItemIds = await getWorkItemIdsByJobUpdateAgeAndStatus(
+          db, notUpdatedForMinutes, jobStatus, startingId, batchSize,
+        );
+        if (workItemIds.length > 0) {
+          const numItemsDeleted = await deleteWorkItemsById(db, workItemIds);
+          totalDeleted += numItemsDeleted;
+          this.logger.info(`Work reaper removed ${numItemsDeleted} work items, starting id: ${startingId}.`);
+          startingId = Math.max(...workItemIds);
```

Comment:
not a big deal, but you don't need `Math.max` here since `workItemIds` is already sorted  - the max is always the last one.

Reply:
This was my defensive coding kicked in. Agreed.